src/cargo/core/compiler/mod.rs

```diff
@@ -696,16 +696,14 @@ fn prepare_rustc(build_runner: &BuildRunner<'_, '_>, unit: &Unit) -> CargoResult
         let tmp = build_runner.files().layout(unit.kind).prepare_tmp()?;
         base.env("CARGO_TARGET_TMPDIR", tmp.display().to_string());
     }
-    if build_runner.bcx.gctx.nightly_features_allowed {
-        // This must come after `build_base_args` (which calls `add_path_args`) so that the `cwd`
-        // is set correctly.
-        base.env(
-            "CARGO_RUSTC_CURRENT_DIR",
-            base.get_cwd()
-                .map(|c| c.display().to_string())
-                .unwrap_or(String::new()),
-        );
-    }
+    // This must come after `build_base_args` (which calls `add_path_args`) so that the `cwd`
+    // is set correctly.
+    base.env(
+        "CARGO_RUSTC_CURRENT_DIR",
+        base.get_cwd()
+            .map(|c| c.display().to_string())
+            .unwrap_or(String::new()),
```

Comment:
Wonder if it could just be.

```suggestion
        base.get_cwd().unwrap_or_default(),
```

(and ditto for `CARGO_TARGET_TMPDIR`? though that is not relevant to this PR)